includes/Checker/Checks/Enqueued_Styles_Size_Check.php

```diff
@@ -0,0 +1,265 @@
+
+		foreach ( $this->get_viewable_post_types() as $post_type ) {
+			$posts = get_posts(
+				array(
+					'posts_per_page' => 1,
+					'post_type'      => $post_type,
+					'post_status'    => array( 'publish', 'inherit' ),
+				)
+			);
+
+			if ( ! isset( $posts[0] ) ) {
+				throw new Exception(
+					sprintf(
+						/* translators: %s: The Post Type name. */
+						__( 'Unable to retrieve post URL for post type: %s', 'plugin-check' ),
+						$post_type
+					)
+				);
+			}
+
+			$urls[] = get_permalink( $posts[0] );
+		}
+
+		return $urls;
+	}
+
+	/**
+	 * Amends the given result by running the check for the given URL.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result The check result to amend, including the plugin context to check.
+	 * @param string       $url    URL to run the check for.
+	 *
+	 * @throws Exception Thrown when the check fails with a critical error (unrelated to any errors detected as part of
+	 *                   the check).
+	 *
+	 * @SuppressWarnings(PHPMD.NPathComplexity)
+	 */
+	protected function check_url( Check_Result $result, $url ) {
+		// Reset the WP_Styles instance.
+		unset( $GLOBALS['wp_styles'] );
+
+		// Run enqueue functions wrapped in an output buffer in case of any callbacks printing styles
+		// directly. This is discouraged, but some plugins or themes are still doing it.
+		ob_start();
+		wp_enqueue_scripts();
+		wp_styles()->do_items();
+		wp_styles()->do_footer_items();
+		ob_get_clean();
+
+		$plugin_styles     = array();
+		$plugin_style_size = 0;
+
+		foreach ( wp_styles()->done as $handle ) {
+			$style = wp_styles()->registered[ $handle ];
+
+			if ( ! $style->src || strpos( $style->src, $result->plugin()->url() ) !== 0 ) {
+				continue;
+			}
+
+			// Get size of style src.
+			$style_path = str_replace( $result->plugin()->url(), $result->plugin()->path(), $style->src );
+			$style_size = function_exists( 'wp_filesize' ) ? wp_filesize( $style_path ) : filesize( $style_path );
+
+			// Get size of additional inline styles.
+			if ( ! empty( $style->extra['after'] ) ) {
+				foreach ( $style->extra['after'] as $extra ) {
+					$style_size += ( is_string( $extra ) ) ? mb_strlen( $extra, '8bit' ) : 0;
+				}
+			}
+
+			if ( ! empty( $style->extra['before'] ) ) {
+				foreach ( $style->extra['before'] as $extra ) {
+					$style_size += ( is_string( $extra ) ) ? mb_strlen( $extra, '8bit' ) : 0;
+				}
+			}
+
+			$plugin_styles[]    = array(
+				'path' => $style_path,
+				'size' => $style_size,
+			);
+			$plugin_style_size += $style_size;
+		}
+
+		if ( $plugin_style_size > $this->threshold_size ) {
+			foreach ( $plugin_styles as $plugin_style ) {
+				$this->add_result_warning_for_file(
+					$result,
+					sprintf(
+						'This style has a size of %1$s which in combination with the other styles enqueued on %2$s exceeds the style size threshold of %3$s.',
+						size_format( $plugin_style['size'] ),
+						$url,
+						size_format( $this->threshold_size )
+					),
```

Comment:
I just realized `Enqueued_Scripts_Size_Check` has the same flaw, so would be great to make the string translatable there as well.

```suggestion
					sprintf(
						/* translators: 1: style file size. 2: tested URL. 3: threshold file size. */
						__( 'This style has a size of %1$s which in combination with the other styles enqueued on %2$s exceeds the style size threshold of %3$s.', 'plugin-check' )
						size_format( $plugin_style['size'] ),
						$url,
						size_format( $this->threshold_size )
					),
```